tests/dotnet/UnitTests/XcodeProjectTests.cs

```diff
@@ -0,0 +1,378 @@
+			File.WriteAllText (proj, newProjContent);
+
+			var xcodeProjName = "TemplateFx";
+			var xcodeProjDirSrc = Path.Combine (XCodeTestProjectDir, xcodeProjName);
+			var xcodeProjDirDest = Cache.CreateTemporaryDirectory ($"{TestName}XcProj");
+			FileHelpers.CopyDirectory (xcodeProjDirSrc, xcodeProjDirDest);
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest, $"{xcodeProjName}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName },
+				});
+
+			DotNet.AssertBuild (proj);
+			foreach (var platform in enabledPlatforms) {
+				AssertXcFrameworkOutput (platform, testDir, xcodeProjName);
+			}
+		}
+
+		[Test]
+		public void InvalidItemErroriOS ()
+		{
+			var platform = ApplePlatform.iOS;
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+
+			var testDir = Cache.CreateTemporaryDirectory (TestName);
+			var proj = Path.Combine (testDir, $"{TestName}.csproj");
+			DotNet.AssertNew (testDir, "ioslib");
+
+			var xcodeProjName = "TemplateFx";
+			var invalidXcodeProjPath = Path.Combine (testDir, "invalid", "path", $"{xcodeProjName}.xcodeproj");
+			AddXcodeProjectItem (proj, invalidXcodeProjPath,
+				new Dictionary<string, string> {
+					{ "SchemeName", xcodeProjName },
+				});
+
+			var rv = DotNet.AssertBuildFailure (proj);
+			var errors = BinLog.GetBuildLogErrors (rv.BinLogPath).Select (e => e.Message).ToArray ();
+			var expectedError = $"The Xcode project item: '{invalidXcodeProjPath}' could not be found. Please update the 'Include' value to a path containing a valid '.xcodeproj' file.";
+			Assert.Contains (expectedError, errors, $"Expected error message '{expectedError}' was not found in build log.");
+		}
+
+		[Test]
+		public void InvalidSchemeErroriOS ()
+		{
+			var platform = ApplePlatform.iOS;
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+
+			var testDir = Cache.CreateTemporaryDirectory (TestName);
+			var proj = Path.Combine (testDir, $"{TestName}.csproj");
+			DotNet.AssertNew (testDir, "ioslib");
+
+			var xcodeProjName = "TemplateFx";
+			var invaldSchemeName = "InvalidScheme";
+			var xcodeProjDirSrc = Path.Combine (XCodeTestProjectDir, xcodeProjName);
+			var xcodeProjDirDest = Cache.CreateTemporaryDirectory ($"{TestName}XcProj");
+			FileHelpers.CopyDirectory (xcodeProjDirSrc, xcodeProjDirDest);
+
+			AddXcodeProjectItem (proj, Path.Combine (xcodeProjDirDest, $"{xcodeProjName}.xcodeproj"),
+				new Dictionary<string, string> {
+					{ "SchemeName", invaldSchemeName },
+				});
+
+			var rv = DotNet.AssertBuildFailure (proj);
+			var expectedErrorContent = $"xcodebuild: error: The project named \"{xcodeProjName}\" does not contain a scheme named \"{invaldSchemeName}\".";
+			var doesContainErrorContent = BinLog.GetBuildLogErrors (rv.BinLogPath).Select (e => e.Message).Any (e => e is not null && e.Contains (expectedErrorContent));
+			Assert.IsTrue (doesContainErrorContent, $"Expected error content '{expectedErrorContent}' was not found in any error message.");
```

Comment:
This one's a bit more complicated since it's a "Contains" comparison instead of "Equals", but there's still a way:

```suggestion
			var expectedErrorContent = $"xcodebuild: error: The project named \"{xcodeProjName}\" does not contain a scheme named \"{invaldSchemeName}\".";
			var errors = BinLog.GetBuildLogErrors (rv.BinLogPath);
			AssertErrorMessages (errors,
				new Func<string, bool> [] {
					(msg) => msg?.Contains (expectedErrorContent) == true
				},
				Func<string, bool> [] {
					(msg) => msg)
				}
			);
```